conans/util/files.py

```diff
@@ -334,3 +334,38 @@ def exception_message_safe(exc):
         return str(exc)
     except:
         return decode_text(repr(exc))
+
+
+def merge_directories(src, dst, excluded=None, symlinks=True):
+    src = os.path.normpath(src)
+    dst = os.path.normpath(dst)
+    excluded = excluded or []
+    excluded = [os.path.normpath(entry) for entry in excluded]
+
+    def is_excluded(origin_path):
+        if origin_path == dst:
+            return True
+        rel_path = os.path.normpath(os.path.relpath(origin_path, src))
+        if rel_path in excluded:
+            return True
+        return False
+
+    for src_dir, dirs, files in walk(src, followlinks=True):
+        if is_excluded(src_dir):
+            dirs[:] = []
+            continue
+
+        # Overwriting the dirs will prevents walk to get into them
+        files[:] = [d for d in files if not is_excluded(os.path.join(src_dir, d))]
+
+        dst_dir = os.path.normpath(os.path.join(dst, os.path.relpath(src_dir, src)))
+        if not os.path.exists(dst_dir):
```

Comment:
`os.path.isdir`? also, what if it exists, but as file?

Reply:
Agree with you, it deserves another PR, in this one I don't want to touch anything related to functionality. Also this PR is just a _draft_ and probably it won't get merged, I'm thinking about dividing it into two PR (this one has changes affecting the `source` command and other targeting the `binary_installer`, I prefer not to mix)